### PR TITLE
aodn_wave - create folder and files and basic content

### PR DESCRIPTION
**Pull request by bpasquer**

### CODE REVIEW

File: aodndata/aodn_wave/aodn_wave.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        fields = get_pattern_subgroups_from_string(file_basename, DTA_NZ_WAVEBUOY)
+        product_dir = fields['site_name']
+
+    elif DES_QLD_WAVEBUOY.match(file_basename):
+        data_base_dir = os.path.join(DES_QLD_DIR, WAVEBUOY_DIR, mode_dir)
+        fields = get_pattern_subgroups_from_string(file_basename, DES_QLD_WAVEBUOY)
+        product_dir = fields['site_name']
+
+    elif MHL_WAVEBUOY.match(file_basename):
+        data_base_dir = os.path.join(NSW_DPE_DIR, MHL_DIR, WAVEBUOY_DIR, mode_dir)
```
Comment (lbesnard) [inline aodndata/aodn_wave/aodn_wave.py:127]: is this normal?
```
 os.path.join(NSW_DPE_DIR, MHL_DIR
```

**Comment by lbesnard** [comment]
@ana-berger replace  with 
```python
DATA_MODE_REGEX = re.compile(f"""
                           (?P<institude_code>({INSTITUDE_CODES}))_
                           (?P<nc_time_cov_start>[0-9]{8})_
                           (?P<site_name>(.*))_(?P<mode>RT|DM)_WAVE-PARAMETERS_END-
                           (?P<nc_time_cov_end>[0-9]{8})\.nc$
                           """.format(INSTITUDE_CODES=INSTITUDE_CODES), re.VERBOSE)
```

re.compile(r
To
re.compile(f

**Comment by ana-berger** [comment]
unit_tests will still be improved (working on that)

**Comment by ana-berger** [comment]
@bpasquer tests still not finished, but just thought would be good to share with you already

**Review by ana-berger — APPROVED** [review]

**Pull request merged**